pkg/config/config.go

```diff
@@ -129,3 +129,26 @@ func (c *Config) LoadIncludes(ctx context.Context, client scm.Client) error {
 
 	return nil
 }
+
+// Merge merges the other config into the current config
+func (c *Config) Merge(other *Config) *Config {
+	if other == nil {
+		return c
+	}
+
+	if other.DryRun != nil {
+		c.DryRun = other.DryRun
+	}
+
+	c.IgnoreActivityFrom.IsBot = other.IgnoreActivityFrom.IsBot
+	c.IgnoreActivityFrom.Usernames = append(c.IgnoreActivityFrom.Usernames, other.IgnoreActivityFrom.Usernames...)
+	c.IgnoreActivityFrom.Emails = append(c.IgnoreActivityFrom.Emails, other.IgnoreActivityFrom.Emails...)
+
+	c.Actions = append(c.Actions, other.Actions...)
+	c.Labels = append(c.Labels, other.Labels...)
+
+	// don't have to worry about duplication here, it is handled when loading the includes
+	c.Includes = append(c.Includes, other.Includes...)
```

Comment:
[optional suggestion] It looks like merging will always append to (and never replace/overwrite) any options that are lists.  I think that's totally fine, but should we mention this behavior in `docs/configuration.md`? Just thinking a user might find this behavior surprising if they're trying to replace the current list with something completely different.

Reply:
yup we should be more explicit in the docs about what merging means, since its less of overriding the global config, and more of merging it on top. ideally there should be a set of actions ran on every repository which are based in the global config, but if someone wants to extend with their own repository specific actions, they don't have to include what was set globally to ensure they continue to get the global actions.